### ISSUE TITLE
support to torch.FloatTensor.view

### ISSUE THREAD

**Issue by EderSantana** [twitter/torch-autograd#92]
Hi,

Thanks for this really interesting module!

I'd like to try a contribution to torch-autograd. `torch.FloatTensor.view` isn't supported yet but I believe it can be pretty useful. 
My first goal was to calculate a set of pairwise distances between the rows of two matrices. In python that would be
```python
sq = ((X[:, None, :] - Z)**2).sum(-1)
```
I was trying the same in Torch with following code:
```lua
local x = X:view(X:size(1), 1, X:size(2))
 x = x:expand(X:size(1), X:size(2), X:size(2))
local z = Z:view(Z:size(1), Z:size(2), 1)
z = z:expandAs(x)
local diffs = torch.sum(x - z, 3)
local sq = torch.cmul(diffs, diffs)
```

But autograd doesn't support `torch.view` yet. But I believe that `view` only reshapes the gradient, without needing any extra operations. The same should happen to `expand`. 

So, my question is, could guys either tell me how I can contribute with support to `view` or suggest me another way to get the `sq` matrix above?

**Comment by soumith** [comment]
I hate doing this on issues, but was gonna use autograd for a new project to try out, :view is a function that i use every 3 lines of my code

**Comment by EderSantana** [comment]
Sorry, I didn't get what you hate, @soumith. Did I say something wrong or posted something in a wrong place?

**Comment by soumith** [comment]
No was just saying that view should be supported

**Comment by EderSantana** [comment]
They actually seem to support `view`:

https://github.com/twitter/torch-autograd/blob/master/src/gradfuns.lua#L358-L362

Why did I get that error then?

**Comment by EderSantana** [comment]
so I guess my problem was with the way that functions are overloaded in Torch. We have autograd for `torch.view(x,...)`, but we can't use `x:view(...)`.

**Comment by alexbw** [comment]
Hey all, thanks for the issue. I'll look into it on Monday.

**Comment by alexbw** [comment]
@EderSantana @soumith, added tests for `torch.view` and `torch.expand` for floating point, both work. I'll look into getting the `x:view()` syntax working now.

**Comment by alexbw** [comment]
Ok, now supported in 255fb95
Any other `x:blah()` methods you're missing?

**Comment by alexbw** [comment]
@EderSantana (see the diff for details on how the gradients were added)[https://github.com/twitter/torch-autograd/commit/255fb9533c4d9bfb79a9660e441952774b8173b0]. Happy to discuss in Slack or further over email if you'd like more explanation, in case you want to contribute in the future.

**Comment by EderSantana** [comment]
whoa! that is awesome. I got my method to work with `torch.view` but that makes it even more elegant and less verbose to write. Tkx a lot!

**Comment by alexbw** [comment]
Glad it works!
